design/director.md

```diff
@@ -0,0 +1,104 @@
+# Director (Build Flow) 
+
+## Context
+
+Running a user's commands inside their container is _the_ point of Screwdriver.  This document describes the expected control flow of a build, specifically how each component interacts with eachother.
+
+## Status
+
+ - 14 September 2017: Proposal submitted
+
+## Proposal
+
+A single interface to launch and manage the commands a user expects to run inside a container that we don't control.
+
+```bash
+$ sd_director --api-uri http://api.screwdriver.cd --build-id 12345 --build-jwt {...}
+```
+
+Here is the high-level overview:
+
+ 1. **Initialization** - Start required services and load information about the build.
+ 2. **Setup** - Ensure the environment is configured and loaded (git clone, environment variables, etc.).
+ 3. **User Steps** - Run all user steps inside a single shell.
+ 4. **Teardown** - Capture artifacts and publish final status.
+
+## Details
```

Comment:
Firstly, I think this is a really good writeup of the direction for the launcher. I'd like a visual diagram of all the shells lifecycles. 

Secondly, and this may be diving too deep into implementation details, but it would be good to talk a little about how we share environment between these shells. For example, if bookends initialize environment variables, those should be available in the user steps, even though it is a different shell.

Reply:
Good idea!  I added a visual diagram for the lifecycles.

Regarding environment variables, I was going to document that heavily in the Bookends doc, but I added a line item on how they get shared here.